WcaOnRails/app/models/competition.rb

```diff
@@ -305,6 +305,18 @@ def events
     eventSpecs.split.map { |e| Event.find_by_id(e.split("=")[0]) }.sort_by &:rank
   end
 
+  def has_event?(event)
+    self.events.include?(event)
+  end
+
+  def belongs_to_region?(region)
+    (self.countryId == region) || (self.country.continentId == region)
+  end
+
+  def contains?(search_param)
+    (name.downcase.include? search_param.downcase) || (cityName.downcase.include? search_param.downcase) || (venue.downcase.include? search_param.downcase)
```

Comment:
Lol that use of parens. A more conventional way of writing this would be:

```ruby
name.downcase.include?(search_param.downcase) || cityName.downcase.include?(search_param.downcase) || venue.downcase.include?(search_param.downcase)
```